### ISSUE TITLE
SOLVER_FLAG_FOCUS_BEST behavior 

### ISSUE THREAD

**Issue by EricAustinYu** [openSUSE/libsolv#314]
I was trying a few things out with SOLVER_FLAG_FOCUS_BEST 

```
repo system 0 testtags <inline>
#>=Pkg: b 1 1 x86_64
repo available 0 testtags <inline>
#>=Pkg: a 2 1 x86_64
#>=Pkg: b 2 1 x86_64
system x86_64 rpm system
solverflags focusbest
job install name a
result transaction,problems <inline>
```

the result is

```
install a-2-1.x86_64@available
upgrade b-1-1.x86_64@system b-2-1.x86_64@available
```

Is it expected that "b" gets upgraded as well when this flag is set even though it's unrelated to installing "a"?

If this is expected, is there another way to make installing "c" also upgrade "d" in the following example without upgrading "b"?

```
repo system 0 testtags <inline>
#>=Pkg: d 1 1 x86_64
#>=Pkg: b 1 1 x86_64
repo available 0 testtags <inline>
#>=Pkg: c 2 1 x86_64
#>=Req: d
#>=Pkg: d 2 1 x86_64
#>=Pkg: b 2 1 x86_64
system x86_64 rpm system
solverflags focusbest
job install name c
result transaction,problems <inline>
```

Thanks!

**Comment by ignatenkobrain** [comment]
I think this is expected. IIRC it has been added to match yum's way of operating: install whatever you want and also update unrelated stuff.

Why do you want to update `d`, but not `b` in this case?

**Comment by Conan-Kudo** [comment]
@EricAustinYu I think you want forcebest flag for a specific package, rather than solver flag focusbest. That way you can only force specific things up to the latest.

**Comment by mlschroe** [comment]
No, it should not update. It's a bug.

**Comment by mlschroe** [comment]
Fixed with commit d3124f5cdbb7fc00d7c204e43b7f94d807c56e91

Thanks for the report!
